keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1138 @@
+
+        // IsDefaultGW6 is a flag indicating this PodNetwork will hold the IPv6 Default
+        // Gateway inside the Pod. Only one Network can have this flag set to True.
+        // This field functionality is dependent on the implementation and its support
+        // for it.
+        //
+        // +optional
+        IsDefaultGW6 bool `json:"isDefaultGW6,omitempty"`
+}
+```
+
+#### Static validations
+We will perform static validation of the provided spec in the API Server,
+alongside the current Pod spec checks. These errors will be provided to the user
+immediately when they try to create Pod.
+This will include:
+* Ensure a single Pod references a given PodNetwork only 1 time
+* IsDefaultGW4 and IsDefaultGW6 uniqueness for “true” value across multiple
+“Network” objects
+* InterfaceName uniqueness across multiple “Network” objects
+* InterfaceName naming constraints for Linux and Windows
+* Ensure Network objects are not specified when hostNetwork field is set
+
+The 1 PodNetwork per Pod restriction is coming from our current lack of details
+for future Multi-Networking requirements (e.g. Service support), and can be
+changed in future when we will discuss them.
+
+#### Active validations
+Beside the above (static validations), we will perform additional active
+validations, inside the  scheduler, that require queries for other objects
+(e.g. PodNetwork). These errors will be presented as Pod Events, and the Pod
+will be kept in Pending state until the issues are resolved. We will do the
+following validation:
+* Referenced PodNetwork or PodNetworkAttachment is present
+* Referenced PodNetwork or PodNetworkAttachment is Ready
+* Referenced PodNetworkAttachment, that is referencing a PodNetwork, follows the
+rule of: single Pod references a given PodNetwork only 1 time
+
+#### Auto-population
+When networks field is not set, and hostNetwork is not set, we will
+auto-populate this field with following values:
+```yaml
+networks:
+- podNetworkName: default
+```
+
+This is to ensure backward compatibility for clusters that will not use
+PodNetwork explicitly.
+
+#### Status
+All the IP addresses for attached PodNetworks will be present in the
+Pod.PodStatus.PodIPs list. To properly identify which IP belongs to what
+PodNetwork, we will expand the PodIP struct to include the name of PodNetwork
+the specific IP belongs to. This list will allow only 1 IP address per family
+(v4, v6) per PodNetwork. The Pod.PodStatus.PodIP behavior will not change.
+Proposed changes below:
+```go
+// IP address information for entries in the (plural) PodIPs field.
+// Each entry includes:
+//
+//      IP: An IP address allocated to the pod.
+//      PodNetworkName: Name of the PodNetwork the IP belongs to.
+//      InterfaceName: Name of the network interface inside the Pod.
+type PodIP struct {
+        // ip is an IP address (IPv4 or IPv6) assigned to the pod
+        IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
```

Comment:
I assume this is the "key" of the struct. This means that all secondary networks will need to actually report IP addresses.

How will this work for L2 secondary networks ? That's a common virt use case. 

Likewise for DPDK applications.

Reply:
Maybe I am not fully aware, but I dont think this struct has a "key", what do you have in mind here?

And the idea here is that not every PodNetwork has to report IP back.

Reply:
> And the idea here is that not every PodNetwork has to report IP back.

Perfect. The scenarios I mentioned above should be OK as long as this is followed.

Thanks.